### PR TITLE
Binning

### PR DESCRIPTION
**Pull request by juyanmei**
Adding dastools which can integrates the results of a flexible number of binning algorithms to calculate an optimized, non-redundant set of bins from a single assembly.
Now it just can integrate the resulits of metabat2 and maxbin2.

### CODE REVIEW

File: metapi/rules/binning.smk [hunk 1]
```diff
@@ -148,12 +150,78 @@ if config["params"]["binning"]["maxbin2"]["do"]:
...
+            -c {output.bins_dir}/scaftigs.fasta \
+            -o {params.bin_prefix} \
+            --write_bins 1 \
+            --threads {threads}
+
+            import glob
+            comp_list = []
+            comp_list = glob.glob("{params.bin_prefix}_DASTools_bins/"+"*.fa")
+            if len(comp_list) != 0: 
+                cp -r {params.bin_prefix}_DASTool_bins/*.fa {output.bins_dir}
```
Comment (alienzj) [inline metapi/rules/binning.smk]: bin prefix issues

**Review by alienzj — CHANGES_REQUESTED** [review]

File: metapi/rules/binning.smk [hunk 2]
```diff
@@ -148,12 +150,78 @@ if config["params"]["binning"]["maxbin2"]["do"]:
...
+            -e fa > {params}.maxbin2.scaffolds2bin.tsv
+
+            pigz -dcp2 {input.scaftigs} > {output.bins_dir}/scaftigs.fasta
+            DAS_Tool \
+            -i {params}.maxbin2.scaffolds2bin.tsv,{params}.metabat2.scaffolds2bin.tsv \
+            -l maxbin,metabat \
+            -c {output.bins_dir}/scaftigs.fasta \
+            -o {params.bin_prefix} \
+            --write_bins 1 \
+            --threads {threads}
```
Comment (alienzj) [inline metapi/rules/binning.smk]: Can we add ` --search_engine` ?

**Review by alienzj — CHANGES_REQUESTED** [review]

File: metapi/rules/binning.smk (hunk 2 above)
Comment (juyanmei) [reply metapi/rules/binning.smk]: Sure, wait a minute, I'll test it.

File: metapi/rules/binning.smk [hunk 3]
```diff
@@ -167,45 +166,72 @@ if config["params"]["binning"]["dastools"]["do"]:
...
                 config["output"]["binning"],
-                "bins/{sample}.{assembler}.out/dastools/{sample}.{assembler}.dastools.bin")
+                "bins/{sample}.{assembler}.out/dastools/{sample}.{assembler}")
         threads:
             config["params"]["binning"]["threads"]
         shell:
             '''
             mkdir -p {output.bins_dir}
+
             Fasta_to_Scaffolds2Bin.sh \
```
Comment (juyanmei) [inline metapi/rules/binning.smk]: Add code to determine if the binning result exists.

**Pull request merged**